test/harness/asyncHelpers-throwsAsync-func-bad-thenable.js

```diff
@@ -0,0 +1,34 @@
+// Copyright (C) 2024 Julián Espina. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+description: |
+    assert.throwsAsync returns a promise that rejects if func returns a bad thenable.
+flags: [async]
+includes: [asyncHelpers.js]
+---*/
+
+asyncTest(async function() {
+  var caught = false;
+  var func = function () {
+    return {
+      then: function () {},
+    };
+  };
+  const p = assert.throwsAsync(TypeError, func);
+  assert(p instanceof Promise, "assert.throwsAsync should return a promise");
```

Comment:
Yeah, that sounds fair. However, I was thinking of ways to test this and maybe it's possible to test with something like:
```js
assert.throwsAsync(TypeError, func).then(
  function() { $DONE("the promise should not be resolved") },
  function() { $DONE("the promise should not be resolved") },
);
$DONE();
```

Though, I'm still unsure if we can assume that implementations will keep running jobs even if the `print` function already printed `Test262:AsyncTestComplete`, but that's maybe something that other tests already assume?

Reply:
> Though, I'm still unsure if we can assume that implementations will keep running jobs even if the `print` function already printed `Test262:AsyncTestComplete`, but that's maybe something other tests already assume?

Presumably. If we're going to cover this, I guess it would be modeled on something like [test/harness/asyncHelpers-asyncTest-returns-undefined.js](https://github.com/tc39/test262/blob/main/test/harness/asyncHelpers-asyncTest-returns-undefined.js) and just overwrite `$DONE` plus document a timeliness assumption:
```js
var realDone = $DONE;
var doneArgs = [];
globalThis.$DONE = function (arg) {
  doneArgs.push(arg);
};

function delay() {
  var later = Promise.resolve();
  for (var i = 0; i < 100; i++) {
    later = later.then();
  }
  return later;
}

(async function () {
  // Spy on the promise returned by an invocation of assert.throwsAsync
  // with a function that returns a thenable which never settles.
  var neverSettlingThenable = { then: function() {} };
  var p = assert.throwsAsync(TypeError, function() { return neverSettlingThenable; });
  p.then($DONE, $DONE);
})()
  // Give it a long time to try.
  .then(delay, delay)
  .then(function () {
    assert.compareArray(doneArgs, [], "$DONE must not have be called");
  })
  .then(realDone, realDone);
```

Reply:
Sounds good! I'll change the test